src/panels/config/users/dialog-add-user.ts

```diff
@@ -0,0 +1,238 @@
+import "@material/mwc-button";
+import "@polymer/paper-spinner/paper-spinner";
+
+import "../../../components/ha-dialog";
+import "../../../resources/ha-style";
+import {
+  LitElement,
+  html,
+  TemplateResult,
+  customElement,
+  property,
+  PropertyValues,
+} from "lit-element";
+import { HomeAssistant } from "../../../types";
+import { PolymerChangedEvent } from "../../../polymer-types";
+import { AddUserDialogParams } from "./show-dialog-add-user";
+import {
+  User,
+  SYSTEM_GROUP_ID_USER,
+  SYSTEM_GROUP_ID_ADMIN,
+  GROUPS,
+  createUser,
+  deleteUser,
+} from "../../../data/user";
+import { createAuthUser } from "../../../data/auth";
+
+@customElement("dialog-add-user")
+export class DialogAddUser extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() private _loading = false;
+  // Error message when can't talk to server etc
+  @property() private _error?: string;
+  @property() private _params?: AddUserDialogParams;
+  @property() private _name?: string;
+  @property() private _username?: string;
+  @property() private _password?: string;
+  @property() private _group?: string;
+
+  public showDialog(params: AddUserDialogParams) {
+    this._params = params;
+    this._name = "";
+    this._username = "";
+    this._password = "";
+    this._group = SYSTEM_GROUP_ID_ADMIN;
+    this._error = undefined;
+    this._loading = false;
+  }
+
+  protected firstUpdated(changedProperties: PropertyValues) {
+    super.firstUpdated(changedProperties);
+    this.addEventListener("keypress", (ev) => {
+      if (ev.keyCode === 13) {
+        this._createUser(ev);
+      }
```

Comment:
We should also prevent default since we handled event.
```suggestion
      if (ev.keyCode === 13) {
        ev.preventDefault();
        this._createUser(ev);
      }
```

Reply:
`this._createUser` does that